src/main/java/com/mineboundteam/minebound/event/ModEvents.java

```diff
@@ -0,0 +1,126 @@
+import com.mineboundteam.minebound.registry.config.ArmorConfigRegistry;
+import net.minecraft.tags.ItemTags;
+import com.mineboundteam.minebound.MineBound;
+import com.mineboundteam.minebound.mana.PlayerMana;
+import com.mineboundteam.minebound.mana.PlayerManaProvider;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.damagesource.DamageSource;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.EquipmentSlot;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.Item;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.Items;
+import net.minecraftforge.common.capabilities.RegisterCapabilitiesEvent;
+import net.minecraftforge.event.AttachCapabilitiesEvent;
+import net.minecraftforge.event.TickEvent;
+import net.minecraftforge.event.entity.player.PlayerEvent;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import net.minecraftforge.fml.LogicalSide;
+import net.minecraftforge.fml.common.Mod;
+
+import java.util.*;
+
+@Mod.EventBusSubscriber(modid = MineBound.MOD_ID)
+public class ModEvents {
+    @SubscribeEvent
+    public static void onAttachCapabilitiesPlayer(AttachCapabilitiesEvent<Entity> event){
+        if(event.getObject() instanceof Player){
+            if(!event.getObject().getCapability(PlayerManaProvider.PLAYER_MANA).isPresent()){
+                event.addCapability(new ResourceLocation(MineBound.MOD_ID, "properties"), new PlayerManaProvider());
+            }
+        }
+    }
+
+    @SubscribeEvent
+    public static void onPlayerCloned(PlayerEvent.Clone event){
+        if(event.isWasDeath()){
+            event.getOriginal().getCapability(PlayerManaProvider.PLAYER_MANA).ifPresent(oldStore ->
+                event.getOriginal().getCapability(PlayerManaProvider.PLAYER_MANA).ifPresent(newStore ->
+                    newStore.copyFrom(oldStore)
+                )
+            );
+        }
+    }
+
+    @SubscribeEvent
+    public static void onRegisterCapabilities(RegisterCapabilitiesEvent event){
+        event.register(PlayerMana.class);
+    }
+
+    @SubscribeEvent
+    public static void onPlayerTick(TickEvent.PlayerTickEvent event){
+        if(event.side == LogicalSide.SERVER && event.player.level.getDayTime() % 10 == 0 && event.phase == TickEvent.Phase.START) {
+            event.player.getCapability(PlayerManaProvider.PLAYER_MANA).ifPresent(mana -> {
+                int manaBoost = 0;
+                int recBoost = 0;
+                Random rndm = new Random();
+
+                boolean armorSet = true;
+                ArmorTier tier = null;
+                List<ItemStack> mArmors = new ArrayList<>();
+
+                // check each armor slot and get their bonuses to recovery and cap
+                for (EquipmentSlot slot : EnumSet.of(EquipmentSlot.HEAD, EquipmentSlot.CHEST, EquipmentSlot.LEGS, EquipmentSlot.FEET)) {
+                    var item = event.player.getItemBySlot(slot).getItem();
```

Comment:
We probably shouldn't use `var`. For consistency and there's no reason not to type all these variables strictly.

Reply:
I was gonna say this, but I did look it up and the types are resolved at compile time for java, so the only thing it affects is readability (and I use var all the time in C# so I don't particularly mind)

Reply:
Readability is probably an important aspect since someone with no knowledge of this code may have to modify it. So, strictly typing will help.